network-api/networkapi/templates/donate/fragments/formassembly_body.html

```diff
@@ -0,0 +1,168 @@
+{% load i18n %}
+
+{% comment %}
+*** IMPORTANT ***
+
+The code below has been copied directly over from FormAssembly. 
+
+Please follow the steps below every time after you paste a new version of FormAssembly code.
+
+1. Replace everything starting from <!-- FORM: BODY SECTION -->
+
+2. Run `inv format-html` in terminal
+
+3. Add nonce="{{ csp_nonce }}" to all script tags in this file
+
+4. Update the revision note on the next line. Revision number can be found on https://mozillafoundation.tfaforms.net/versions/index/12
+
+The code snippet below is based on FormAssembly form revision #40
+
+{% endcomment %}
+
+<!-- FORM: BODY SECTION -->
+<div class="wFormContainer" style="max-width: 100%; width:auto;">
+    <div class="wFormHeader"></div>
+    <div class=""><div class="wForm" id="12-WRPR" dir="ltr">
+        <div class="codesection" id="code-12"><script nonce="{{ request.csp_nonce }}">
+// From Mel A at Thunderbird. 06.09.2023. Edited by Mia to change adding en- locales to map and defaulting to 'Other'. Approved by Mel.
+            window.addEventListener('DOMContentLoaded', function () {
+    // Retrieve the Language alias (locale field) on the form.
+                const varLocale = document.getElementById('tfa_72');
+
+    // Exit if we can't find the locale selection dropdown
+                if (!varLocale) {
+                    return;
+                }
+
+    // Retrieve the url, and attempt to get the first parameter (locale)
+                const url = new URL(window.location);
+                const locale = url.pathname.split('/')[1] || null;
+
+    // If locale does not exist in the url, then there's nothing more to do.
+                if (!locale) {
+                    return;
+                }
+
+    // Match up locale codes with the locale dropdown entries
+                const localeMap = {
+                    'zh-CN': 'tfa_218',
+                    'zh-TW': 'tfa_218',
+                    'cs': 'tfa_219',
+                    'da': 'tfa_220',
+                    'nl': 'tfa_221',
+                    'en-CA': 'tfa_222',
+                    'en-GB': 'tfa_222',
+                    'en-US': 'tfa_222',
+                    'en-ZA': 'tfa_222',
+                    'fr': 'tfa_223',
+                    'de': 'tfa_224',
+                    'it': 'tfa_225',
+                    'ja': 'tfa_226',
+                    'pl': 'tfa_228',
+                    'pt-BR': 'tfa_229',
+                    'ru': 'tfa_230',
+                    'es-MX': 'tfa_231',
+                    'es-AR': 'tfa_231',
+                    'es-ES': 'tfa_231',
+                    'es-CL': 'tfa_231'
+                }
+
+    // Assign the locale, or default to 'Other' if the locale isn't supported.
+                varLocale.value = localeMap[locale] || 'tfa_227';
+            });
+
```

Comment:
Hi @mmmavis just calling out this custom localization JS that I found from Mia and the thunderbird team. I am probably going to have to update this in my localization PR as they are referencing locales we do not use on the foundation site (ex: en-US, es-MX, etc.).